lib/query/token-visitors.js

```diff
@@ -0,0 +1,188 @@
+const { isEmpty, compact, isNil, isFunction, isString, cloneDeep, castArray } = require("lodash");
+const Tokenizer = require("tokenize-this");
+const { operators } = require("../operators");
+
+class TokenListVisitor {
```

Comment:
Name sounds somewhat confusing, it sounds like a very generic visitor, but it actually has a specific purpose; can it be reflected in the name?

Reply:
Yeah, updated to SQLTokenListVisitor.